drivers/SmartThings/zigbee-air-quality-detector/src/MultiIR/init.lua

```diff
@@ -0,0 +1,182 @@
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local clusters = require "st.zigbee.zcl.clusters"
+local capabilities = require "st.capabilities"
+local custom_clusters = require "MultiIR/custom_clusters"
+local cluster_base = require "st.zigbee.cluster_base"
+
+local RelativeHumidity = clusters.RelativeHumidity
+local TemperatureMeasurement = clusters.TemperatureMeasurement
+
+local MultiIR_SENSOR_FINGERPRINTS = {
+  { mfr = "MultiIR", model = "PMT1006S-SGM-ZTN" }--This is not a sleep end device
+}
+
+local function can_handle_MultiIR_sensor(opts, driver, device)
+  for _, fingerprint in ipairs(MultiIR_SENSOR_FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local function send_read_attr_request(device, cluster, attr)
+  device:send(
+    cluster_base.read_manufacturer_specific_attribute(
+      device,
+      cluster.id,
+      attr.id,
+      cluster.mfg_specific_code
+    )
+  )
+end
+
+local function do_refresh(driver, device)
+  device:send(RelativeHumidity.attributes.MeasuredValue:read(device):to_endpoint(0x01))
+  device:send(TemperatureMeasurement.attributes.MeasuredValue:read(device):to_endpoint(0x01))
+  send_read_attr_request(device, custom_clusters.particulate_matter, custom_clusters.particulate_matter.attributes.pm2_5_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.particulate_matter, custom_clusters.particulate_matter.attributes.pm1_0_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.particulate_matter, custom_clusters.particulate_matter.attributes.pm10_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.unhealthy_gas, custom_clusters.unhealthy_gas.attributes.CH2O_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.unhealthy_gas, custom_clusters.unhealthy_gas.attributes.tvoc_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.carbonDioxide, custom_clusters.carbonDioxide.attributes.measured_value)
+end
+
+local LEVEL_TO_NUMBER = {
+  { level = "good", value = 1 },
+  { level = "moderate", value = 2 },
+  { level = "unhealthy", value = 3 }
+}
+
+local cap_index = {
+  carbonDioxide = 1,
+  pm2_5 = 2,
+  pm1_0 = 3,
+  pm10 = 4,
+  tvoc = 5
+}
+
+local cap_save_level = {
+  { level = "good", value = 1 },
+  { level = "good", value = 1 },
+  { level = "good", value = 1 },
+  { level = "good", value = 1 },
+  { level = "good", value = 1 }
+}
```

Comment:
Local variables like this apply to the whole driver, which may host multiple devices.

If you want to save state, use either `device:set_field` or `device:get_latest_state`, which automatically stores the last emitted value for each capability.